### PR TITLE
Feature 71 confirm dialog before auth

### LINKED ISSUE #73: Confirm dialog for the form authentication redirect
As a user, I'd like to see some kind of information and confirm dialog before app redirects me to keycloak site.

### PR DESCRIPTION
**Pull request by EVP1989**
Resolves #73

### CODE REVIEW

File: src/components/generic/auth-redirect-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  /**
+   * Constructor
+   * 
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = { };
+  }
```
Comment (Chi-Chu319) [inline src/components/generic/auth-redirect-dialog.tsx:34]:
```suggestion
  constructor(props: Props) {
    super(props);
    this.state = {};
  }
```

File: src/components/generic/auth-redirect-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+   * Component render method
+   */
+  public render() {
+    const { onConfirm, classes } = this.props;
+    
+    return (
+      <Dialog
+        open={ true }
+        BackdropProps={{ style:{ backgroundColor: "#fff", paddingBottom:"100px" }} }
+      >
```
Comment (Chi-Chu319) [inline src/components/generic/auth-redirect-dialog.tsx:46]:
```suggestion
      <Dialog
        open
        BackdropProps={{ style:{ backgroundColor: "#fff", paddingBottom:"100px" }} }
      >
```

File: src/reducers/auth.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import { AuthAction } from '../actions/auth';
 import { SIGNED_LOGIN, SIGNED_LOGOUT, ANONYMOUS_LOGIN } from '../constants/actionTypes';
 import { KeycloakInstance } from 'keycloak-js';
-import { AccessToken } from '../types';
+import { AccessToken, SignedToken } from '../types';
```
Comment (Chi-Chu319) [inline src/reducers/auth.ts:4]: use double quote

**Review by Chi-Chu319 — CHANGES_REQUESTED** [review]
few suggestions

File: src/components/containers/signed-token-refresh.tsx [hunk 4]
```diff
@@ -42,57 +46,70 @@ class SignedTokenRefresh extends React.Component<Props, State> {
...
 
   /**
    * Component render method
    */
   public render() {
-    if (this.state.error) {
-      return <ErrorDialog error={ this.state.error } onClose={ () => this.setState({ error: undefined }) } />;
+    const { children, signedToken, classes, loginMode } = this.props;
+    const { error } = this.state;
+
```
Comment (anttileppa) [inline src/components/containers/signed-token-refresh.tsx:92]:
```suggestion
```

File: src/components/containers/signed-token-refresh.tsx [hunk 5]
```diff
@@ -42,57 +46,70 @@ class SignedTokenRefresh extends React.Component<Props, State> {
...
+      return (
+        <ConfirmAuthRedirectDialog
+          classes={ classes }
+          onConfirm={ () => this.keycloak.login(Config.getSignedKeycloakLoginOptions(loginMode)) }
+          dialogOpen
+        />
+      );
+    }
+    
+    if (error) {
```
Comment (anttileppa) [inline src/components/containers/signed-token-refresh.tsx:103]: Move this above the logic

**Review by anttileppa — APPROVED** [review]

**Review by Chi-Chu319 — APPROVED** [review]

**Pull request merged**